routes.go

```diff
@@ -95,15 +95,21 @@ type PrimaryMMIn struct {
 	Primary alice.Chain `name:"middleware_primary_metrics"`
 }
 
+type AlternateMMIn struct {
+	fx.In
+	Primary alice.Chain `name:"middleware_alternate_metrics"`
+}
+
 type HealthMMIn struct {
 	fx.In
 	Health alice.Chain `name:"middleware_health_metrics"`
 }
 
 type MetricMiddlewareOut struct {
 	fx.Out
-	Primary alice.Chain `name:"middleware_primary_metrics"`
-	Health  alice.Chain `name:"middleware_health_metrics"`
+	Primary   alice.Chain `name:"middleware_primary_metrics"`
+	Alternate alice.Chain `name:"middleware_alternate_metrics"`
+	Health    alice.Chain `name:"middleware_health_metrics"`
 }
 
 func provideServers() fx.Option {
```

Comment:
In general, try to reduce the amount of unmarshaling via `Provide...` since that leads to some noisy code.  You can clean this up a bit by declaring a `struct` and unmarshaling that once.  Also, `golang` de facto standard is to avoid abbreviations where possible.

```
type SomeConfig struct {
    RequestMaxRetries int `mapstructure:"whatever the config key is"`

    // additional fields ...
}

// then use:
fx.Options(
    arrange.ProvideKey("someConfigKey", SomeConfig{}),
    // other options can now have SomeConfig injected
)

```

Reply:
tracking this work on another ticket https://github.com/comcast-cl/xmidt/issues/447